game/theater/theatergroundobject.py

```diff
@@ -635,3 +682,58 @@ def group_name(self) -> str:
         # Prefix the group names with the side color so Skynet can find them,
         # add to EWR.
         return f"{self.faction_color}|EWR|{super().group_name}"
+
+
+class IadsBuildingGroundObject(BuildingGroundObject):
+    iads_role: IADSRole  # IadsBuilding GO has only one iads_role
+
+    def __init__(
+        self,
+        name: str,
+        group_id: int,
+        preset_location: PresetLocation,
+        control_point: ControlPoint,
+        iads_role: IADSRole,
+    ) -> None:
+        super().__init__(
+            name=name,
+            category=self.iads_category_for(iads_role),
+            group_id=group_id,
+            object_id=0,  # Only 1 building
+            position=preset_location,
+            heading=preset_location.heading,
+            control_point=control_point,
+            dcs_identifier=self.iads_identifier_for(iads_role),
+        )
+        self.original_name = preset_location.original_name  # store original name
+        self.iads_role = iads_role
+
+    def mission_types(self, for_player: bool) -> Iterator[FlightType]:
+        from game.ato import FlightType
+
+        if not self.is_friendly(for_player):
+            yield from [FlightType.STRIKE, FlightType.DEAD]
+
+    @property
+    def connection_range(self) -> Distance:
+        if self.iads_role == IADSRole.ConnectionNode:
+            return Distance(27780)  # 15nm
+        elif self.iads_role == IADSRole.PowerSource:
+            return Distance(64820)  # 35nm
+        return Distance(0)
```

Comment:
This needs more fine tuning. For now it was just based on the needs i had for testing. Need some Input from Campaign Designers :)